lib/User_LDAP.php

```diff
@@ -182,11 +182,18 @@ public function getUsers($search = '', $limit = 10, $offset = 0) {
 			$filter,
 			$this->access->userManager->getAttributes(true),
 			$limit, $offset);
-		$ldap_users = $this->access->ownCloudUserNames($ldap_users);
-		\OCP\Util::writeLog('user_ldap', 'getUsers: '.count($ldap_users). ' Users found', \OCP\Util::DEBUG);
+		$ldap_users2 = $this->access->ownCloudUserNames($ldap_users);
+		\OCP\Util::writeLog('user_ldap', 'getUsers: '.count($ldap_users2). ' Users found', \OCP\Util::DEBUG);
+
+		// sanity check
+		$ldap_users_count = count($ldap_users);
+		$ldap_users2_count = count($ldap_users2);
+		if ($ldap_users_count !== $ldap_users2_count) {
+			\OCP\Util::writeLog('user_ldap', "getUsers: fetched $ldap_users_count users but returned $ldap_users2_count", \OCP\Util::ERROR);
+		}
 
-		$this->access->connection->writeToCache($cachekey, $ldap_users);
-		return $ldap_users;
+		$this->access->connection->writeToCache($cachekey, $ldap_users2);
+		return $ldap_users2;
```

Comment:
It doesn't seem possible. Core asks for 500 users and LDAP is returning less. Returning more information from LDAP is useless at the moment because that info won't be used by core.

On the other hand, core needs to handle other backends. The condition to decide whether keep fetching users or not is pretty clear: if core asks for 500 users and 500 users are returned, there could be more users, otherwise it's the last bunch. This way we don't need to count how many users there are, since this is time-consuming specially for LDAP (we need to fetch all the entries)

In addition, if someone asks core for 500 users it should return 500 users. This means that if the backend returns less, core should ask for more. For LDAP, LDAP will need to fetch the next 500 users even though we might only need a few of them (I don't think we can change the page size during a search) which will require complex handling.